### PR TITLE
NIAD-223: Scaffold of Manager/Translator to be used by further accepta…

### PR DESCRIPTION
**Pull request by mattd-kainos**
…nce stories

### CODE REVIEW

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/DateTimePeriod.java [hunk 1]
```diff
@@ -1,9 +1,6 @@
 package uk.nhs.digital.nhsconnect.nhais.model.edifact;
 
-import lombok.Getter;
-import lombok.NonNull;
-import lombok.RequiredArgsConstructor;
-import lombok.Setter;
+import lombok.*;
```
Comment (bartek-sarul) [inline src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/DateTimePeriod.java]: Not recommended

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/model/fhir/ParameterNames.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package uk.nhs.digital.nhsconnect.nhais.model.fhir;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor @Getter
+public enum ParameterNames {
+    GP_TRADING_PARTNER_CODE("gpTradingPartnerCode"),
```
Comment (bartek-sarul) [inline src/main/java/uk/nhs/digital/nhsconnect/nhais/model/fhir/ParameterNames.java:8]: @grzegorz-robak did that in his PR so we'll have to consolidate those changes into one place

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/DateTimePeriod.java (hunk 1 above)
Comment (mattd-kainos) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/DateTimePeriod.java]: I blame intellij, can revert

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/service/FhirToEdifactService.java [hunk 3]
```diff
@@ -8,28 +8,18 @@
...
-import uk.nhs.digital.nhsconnect.nhais.model.edifact.InterchangeTrailer;
-import uk.nhs.digital.nhsconnect.nhais.model.edifact.MessageHeader;
-import uk.nhs.digital.nhsconnect.nhais.model.edifact.MessageTrailer;
-import uk.nhs.digital.nhsconnect.nhais.model.edifact.NameAndAddress;
-import uk.nhs.digital.nhsconnect.nhais.model.edifact.ReferenceTransactionNumber;
-import uk.nhs.digital.nhsconnect.nhais.model.edifact.ReferenceTransactionType;
-import uk.nhs.digital.nhsconnect.nhais.model.edifact.Segment;
-import uk.nhs.digital.nhsconnect.nhais.model.edifact.SegmentGroup;
-import uk.nhs.digital.nhsconnect.nhais.model.edifact.TranslatedInterchange;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.*;
```
Comment (bartek-sarul) [inline src/main/java/uk/nhs/digital/nhsconnect/nhais/service/FhirToEdifactService.java]: Not recommended

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/translator/FhirToEdifactManager.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    private List<Segment> delegateAcceptance(Parameters parameters) throws FhirValidationException {
+        String acceptanceType = determineAcceptanceType(parameters);
+        if("birth".equals(acceptanceType)) {
+            return acceptanceBirthTranslator.translate(parameters);
+        }
+        return null;
+    }
+
+    private String determineAcceptanceType(Parameters parameters) {
+        return "birth";
```
Comment (bartek-sarul) [inline src/main/java/uk/nhs/digital/nhsconnect/nhais/translator/FhirToEdifactManager.java:42]: Let's make this an enum later

**Review by bartek-sarul — APPROVED** [review]

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/translator/FhirToEdifactManager.java (hunk 4 above)
Comment (mattd-kainos) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/translator/FhirToEdifactManager.java:42]: 👍 its just a placeholder so the example code compiles

**Review by petervdm — APPROVED** [review]

**Pull request merged**